### PR TITLE
Better UI/UX for suggestions for new users (IOS-157)

### PR DESCRIPTION
**Pull request by zeitschlag**
Say hello to the new "Hey, do you already follow there people?"-screen, that appears every time you open the app and don't follow people. And when you press the "Find people to follow"-button on the empty-screen

It uses the new `UserView` for every account, but offers the option to follow them all.

</img></img>

### CODE REVIEW

File: Mastodon/Scene/SuggestionAccount/TableViewCell/SuggestionAccountTableViewCell+ViewModel.swift [hunk 1]
```diff
@@ -138,4 +74,62 @@ extension SuggestionAccountTableViewCell {
         .assign(to: \.isPending, on: viewModel)
         .store(in: &disposeBag)
     }
+
+    class ViewModel {
+        var disposeBag = Set<AnyCancellable>()
```
Comment (DK1DA) [inline Mastodon/Scene/SuggestionAccount/TableViewCell/SuggestionAccountTableViewCell+ViewModel.swift]: Can this be made `private`?
```suggestion
        private var disposeBag = Set<AnyCancellable>()
```

File: Mastodon/Scene/SuggestionAccount/TableViewCell/SuggestionAccountTableViewCell+ViewModel.swift [hunk 2]
```diff
@@ -138,4 +74,62 @@ extension SuggestionAccountTableViewCell {
...
+        
+        @Published public var userIdentifier: UserIdentifier?       // me
+        
+        @Published var avatarImageURL: URL?
```
Comment (DK1DA) [inline Mastodon/Scene/SuggestionAccount/TableViewCell/SuggestionAccountTableViewCell+ViewModel.swift]: Why is this `internal` whereas `authorName` and `authorUsername` are private? Shouldn't it be on the same access level?

**Review by kimar — COMMENTED** [review]
Great work. 👏  I see that your Pr is a draft, just adding a few notes here.

File: Mastodon/Scene/SuggestionAccount/TableViewCell/SuggestionAccountTableViewCell+ViewModel.swift (hunk 2 above)
Comment (zeitschlag) [reply Mastodon/Scene/SuggestionAccount/TableViewCell/SuggestionAccountTableViewCell+ViewModel.swift]: To be honest: I don't know (yet), I didn't write this code, I just moved it around a bit :)

File: Mastodon/Scene/Share/View/TableviewCell/UserTableViewCell+ViewModel.swift [hunk 3]
```diff
@@ -52,20 +54,24 @@ extension UserTableViewCell {
...
             .receive(on: DispatchQueue.main)
-            .sink { [weak self] followed, blocked in
+            .sink { [weak self] followed, requested, blocked in
                 if blocked.contains(user.id) {
                     self?.userView.setButtonState(.blocked)
                 } else if followed.contains(user.id) {
                     self?.userView.setButtonState(.unfollow)
+                } else if requested.contains(user.id) {
+                    self?.userView.setButtonState(.pending)
+                } else if user.locked {
```
Comment (DK1DA) [inline Mastodon/Scene/Share/View/TableviewCell/UserTableViewCell+ViewModel.swift:70]: 🥇 thank you @zeitschlag !

File: Mastodon/Diffable/Search/SearchHistorySection.swift [hunk 4]
```diff
@@ -32,7 +32,12 @@ extension SearchHistorySection {
                 guard let user = item.object(in: context.managedObjectContext) else { return }
                 cell.configure(
                     me: authContext.mastodonAuthenticationBox.authenticationRecord.object(in: context.managedObjectContext)?.user,
-                    viewModel: .init(value: user, followedUsers: authContext.mastodonAuthenticationBox.inMemoryCache.$followingUserIds.eraseToAnyPublisher(), blockedUsers: authContext.mastodonAuthenticationBox.inMemoryCache.$blockedUserIds.eraseToAnyPublisher()),
+                    viewModel: SearchHistoryUserCollectionViewCell.ViewModel(
+                        value: user,
+                        followedUsers: authContext.mastodonAuthenticationBox.inMemoryCache.$followingUserIds.eraseToAnyPublisher(),
+                        blockedUsers: authContext.mastodonAuthenticationBox.inMemoryCache.$blockedUserIds.eraseToAnyPublisher(),
+                        followRequestedUsers: authContext.mastodonAuthenticationBox.inMemoryCache.$followRequestedUserIDs.eraseToAnyPublisher()
```
Comment (DK1DA) [inline Mastodon/Diffable/Search/SearchHistorySection.swift:39]: We'd also need to fetch the pending requests from https://docs.joinmastodon.org/methods/follow_requests/ to pre-seed this e.g. on user switch etc. basically the same as done here: https://github.com/mastodon/mastodon-ios/blob/develop/MastodonSDK/Sources/MastodonCore/Service/AuthenticationService.swift#L50

Comment (zeitschlag) [reply Mastodon/Diffable/Search/SearchHistorySection.swift:39]: Ah, you're right! Good that you saw this 👍  I'll add that 🙂

File: MastodonSDK/Sources/MastodonUI/View/Content/UserView.swift [hunk 5]
```diff
@@ -99,7 +99,13 @@ public final class UserView: UIView {
         label.textColor = .secondaryLabel
         return label
     }()
-    
+
+    private let followButtonWrapper = {
```
Comment (DK1DA) [inline MastodonSDK/Sources/MastodonUI/View/Content/UserView.swift]: 💡  Looks like we could inline this after all...
```suggestion
    private let followButtonWrapper = UIView()
```

File: Mastodon/Scene/SuggestionAccount/SuggestionAccountViewController.swift [hunk 6]
```diff
@@ -168,46 +98,37 @@ extension SuggestionAccountViewController: UITableViewDelegate {
...
-                }
-                cell.stopAnimating()
-            }   // end Task
+extension SuggestionAccountViewController: SuggestionAccountTableViewCellDelegate { }
+
+
+extension SuggestionAccountViewController: SuggestionAccountTableViewFooterDelegate {
+    func followAll(_ footerView: SuggestionAccountTableViewFooter) {
+        viewModel.followAllSuggestedAccounts(self) {
+            OperationQueue.main.addOperation {
```
Comment (DK1DA) [inline Mastodon/Scene/SuggestionAccount/SuggestionAccountViewController.swift]: 💡  No need to change, but I think we're mainly using GCD directly 👇 
```suggestion
            DispatchQueue.main.async {
```

**Review by kimar — APPROVED** [review]
LGTM 👍

File: MastodonSDK/Sources/MastodonUI/View/Content/UserView.swift (hunk 5 above)
Comment (zeitschlag) [reply MastodonSDK/Sources/MastodonUI/View/Content/UserView.swift]: Oh 🙈😅 You're absolutely right!

File: Mastodon/Scene/SuggestionAccount/SuggestionAccountViewController.swift (hunk 6 above)
Comment (zeitschlag) [reply Mastodon/Scene/SuggestionAccount/SuggestionAccountViewController.swift]: Great, that you saw that. I'm the only one using `OperationQueue.main.addOperation`. I changed there in the other places, too 👍

**Pull request merged**